src/app/modules/account/login/login.component.ts

```diff
@@ -17,23 +17,55 @@
 
 import { Component } from '@angular/core';
 import { AuthService } from 'src/app/services/auth.service';
+import { BsModalService, ModalOptions } from 'ngx-bootstrap/modal';
+import { ModalAcceptGenericComponent } from 'src/app/components/modals/modal-accept-generic/modal-accept-generic.component';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss'],
 })
 export class LoginComponent {
-  constructor(private authService: AuthService) { }
+  constructor(private authService: AuthService,
+              private modalService: BsModalService,
+    ) { }
   public readonly title = 'Sign in';
 
   /**
    * Method which triggers when the login FHICT button is pressed.
    */
   public onClickLoginFHICT() {
-    this.authService.login('FHICT');
+    this.loginWithConsent('FHICT');
   }
   public onClickLoginIdentityServer() {
-    this.authService.login();
+    this.loginWithConsent();
+  }
+
+  /**
+   * Checks if the user already has given consent to use their data.
+   * If the user did not consent before a modal is shown to request consent.
+   */
+  private loginWithConsent(provider?: string): void {
+    const privacyConsentKeyName = 'PrivacyConsentGiven';
+    if (localStorage.getItem(privacyConsentKeyName) === 'true') {
+      this.authService.login(provider);
+    } else {
+      // set modal options
+      const modalOptions: ModalOptions = {
+        initialState: {
+          titleText: 'We value your privacy!',
+          mainText: `By clicking accept, you agree to our <a href="/policy" target="_blank">Privacy Policy</a>. You acknowledge that you have read and understand it. If you have questions or concerns you can contact us.`,
```

Comment:
Router link should be used in favor of href

Reply:
I tried this but the browser sanitizes it and removes the link with routerlink.
then i tried to sanitize it as html (that worked) but the link was not clickable.
so we settled on keeping it a href